src/solvers/mp/MasterProblem.cpp

```diff
@@ -0,0 +1,1278 @@
+  }
+}
+
+//------------------------------------------------
+// Solution with rolling horizon process
+//------------------------------------------------
+
+// relax/unrelax the integrality constraints of the variables corresponding to input days
+//
+void MasterProblem::relaxDays(vector<bool> isRelax) {
+	if (isRelaxDay_.empty()) {
+		isRelaxDay_.insert(isRelaxDay_.begin(),pDemand_->nbDays_,false);
+		isPartialRelaxDays_ = true;
+	}
+
+	for (int day=0; day < pDemand_->nbDays_; day++) {
+		isRelaxDay_[day]= isRelaxDay_[day]? true:isRelax[day];
+	}
+
+	pModel_->relaxRotationsStartingFromDays(isRelax);
+}
+
+void MasterProblem::unrelaxDays(vector<bool> isUnrelax) {
+	if (isRelaxDay_.empty()) {
+		isPartialRelaxDays_= false;
+	}
+	else {
+    for (int day = 0; day < pDemand_->nbDays_; day++)
+      isRelaxDay_[day] = isRelaxDay_[day] ? !isUnrelax[day] : false;
+    pModel_->unrelaxRotationsStartingFromDays(isUnrelax);
+  }
+}
+
+// fix/unfix all the variables corresponding to the input vector of days
+void MasterProblem::fixDays(vector<bool> isFix) {
+	// initialize the list of fixed days if empty
+	if (isFixDay_.empty()) {
+		isFixDay_.insert(isFixDay_.begin(),pDemand_->nbDays_,false);
+		isPartialFixDays_ =true;
+	}
+
+	// set the list of fixed day
+	// + forbid the generation of rotations with these starting days
+	for (int day=0; day < pDemand_->nbDays_; day++) {
+		isFixDay_[day]= isFixDay_[day]?true:isFix[day];
+		if (isFixDay_[day]) pPricer_->forbidStartingDay(day);
+	}
+
+	// actually fix the active columns starting with the input days with their current values
+	pModel_->fixRotationsStartingFromDays(isFix);
+}
+void MasterProblem::unfixDays(vector<bool> isUnfix) {
+	// easy to treat the case where no day is fixed yet
+	if (isFixDay_.empty()) {
+		isPartialFixDays_ = false;
+		pPricer_->clearForbiddenStartingDays();
+	}
+	else {
+		// set the list of unfixed day
+		// + authorize the generation of rotations with these starting days
+		for (int day=0; day < pDemand_->nbDays_; day++) {
+			isFixDay_[day]= isFixDay_[day]?!isUnfix[day]:false;
+			if (!isFixDay_[day]) pPricer_->authorizeStartingDay(day);
+		}
+
+		// actually unfix the active columns starting with the input days
+		pModel_->unfixRotationsStartingFromDays(isUnfix);
+	}
+}
+
+// fix/unfix all the variables corresponding to the input vector of nurse ids
+void MasterProblem::fixNurses(vector<bool> isFix){
+	// initialize the list of fixed nurses if empty
+	if (isFixNurse_.empty()) {
+		isFixNurse_.insert(isFixNurse_.begin(),getNbNurses(),false);
+		isPartialFixNurses_ = true;
+	}
+	// set the list of fixed day
+	// + forbid the generation of rotations of the input nurses
+	for (LiveNurse* pNurse: theLiveNurses_){
+		int n = pNurse->id_;
+		isFixNurse_[n] = isFixNurse_[n]?true:isFix[n];
+		if (isFixNurse_[n]) pPricer_->forbidNurse(n);
+	}
+
+	// actually fix the active columns of the input nurses with their current values
+	pModel_->fixRotationsOfNurses(isFix);
```

Comment:
This line and l. 364 seem to be relevant only to RotationMP, but I guess it could be made more general, e.g., with a virtual function that fixes either rotations or rosters depending on the decomposition. Am I wrong?